toqito/states/max_mixed.py

```diff
@@ -68,5 +67,5 @@ def max_mixed(dim: int, is_sparse: bool = False) -> [np.ndarray, dia_array]:
 
     """
     if is_sparse:
-        return 1 / dim * sparse.eye(dim)
+        return 1 / dim * eye_array(dim)
```

Comment:
@purva-thakre I'm unsure if `eye_array` should be used or not? I kept getting the following warning: `This function returns a sparse matrix -- not a sparse array. You are encouraged to use eye_array to take advantage of the sparse array functionality.`

My thought was this is what's wanted based off: https://github.com/vprusso/toqito/issues/528#issuecomment-2041142345

And the data type for both functions kept returning as a matrix I noticed when fixing the imports:

```>>> from toqito.states import max_entangled
>>> max_entangled(2, True, True)
<class 'scipy.sparse._dia.dia_matrix'>
<4x1 sparse matrix of type '<class 'numpy.float64'>'
	with 2 stored elements in COOrdinate format>
```

```
>>> from toqito.states import max_mixed
>>> max_mixed(2, is_sparse=True)
<2x2 sparse matrix of type '<class 'numpy.float64'>'
	with 2 stored elements (1 diagonals) in DIAgonal format>
```

Reply:
Go for it. I think I missed this function when I made the linked comment. 

Feel free to use other instances of `sparse.array` if you need to. Chances are I could have missed other `sparse.matrix` functions.